Shopify/cli
# [Feature] webhook trigger: Improve client secret integration

### WHY are these changes introduced?

Related tasks:
* https://github.com/orgs/Shopify/projects/4831/views/1?pane=issue&itemId=16213451
* https://github.com/orgs/Shopify/projects/4831/views/1?pane=issue&itemId=16450983


### WHAT is this pull request doing?

It renames `--shared-secret` to `--client-secret` (shared-secret still there but showing a deprecation warning)
It allows you to use a secret automatically from the app config:
* .env file if you've previously done `env pull`
* server, by asking about the app (or guessing it because you only have one or because your folder matches any of the existing ones)

A graphic explanation of the new flow:
```mermaid
graph TD
    start[Get Shared-Secret flow] --> flag{--shared-secret}
    flag --> |it's set| usef[Use --shared-secret flag value]
    flag --> |it's not set or empty| man>Prompt: Set manually?]
    man --> |yes| many>Prompt for secret] --> manys[Use secret]
    man --> |no| env{.env & secret present}
    env --> |yes| envyp[/Print: Using .env secret/] --> envyu[Use .env secret]
    env --> |no| autoa(retrieve available apps)
    autoa --> |some options available| autof(get current folder name) --> autofy{folder name in list?} --> |yes| selappr(retrieve app secret) --> autofyp[/Print: Using app secret/] --> autofyu[Use app secret]
    autofy --> |no| only{Only one app}
    only --> |yes| selappr
    only --> |no| selapp>Prompt to select app] --> selappr
    autoa --> |no options| noapp>Prompt for secret] --> noappu[Use secret] 
```

### How to test your changes?

spin up:
```
schema: v1
extend: identity
repos:
  - partners:
      env:
        DISABLE_SHOPIFY_INTERNAL_API_INTERCEPTOR: "1"
  - captain-hook
  - shopify
  - storefront-renderer
  - webhook-receiver
  - web
```

These are the cases I've tested ✅ :

1. try the command with --shared-secret. It should work (do not prompt for secret), but show this warning:

> The flag shared-secret has been deprecated in favor of client-secret and will be eventually deleted.
> Please use --client-secret instead.

2. try the command with --client-secret. It should work without warning
### Post-release steps

3. Try the command without the secret flag. You should be prompted for:
```
?  Do you want to set the client-secret manually?

>  (y) Yes, I'll type it myself
   (n) No, try to get it from the app config
```
4. If you select `(y)` then you should be prompted for secret and it'll be used

The following cases apply when you select `(n)`

5. Create .env file without SHOPIFY_API_SECRET: It should go through the prompt process (see from point 7 on)
6. Create a .env file with SHOPIFY_API_SECRET: It should use the secret stored in that file

7. If you have no app: You should get prompted for secret value (you'll be prompted for org and app selection previously)
9. If you only have one app: Your app secret should be used
10. If you have several apps and one of them matches the name of your current folder: The app secret should be used
11. If you have several apps and none matches the name of your current folder: You will be prompted to select app and its secret will be used

Post-release: Update dev docs

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
